src/js/core/game_state.ts

```diff
@@ -180,39 +169,43 @@ export class GameState {
     /**
      * Should return whether to fade in the game state. This will then apply the right css classes
      * for the fadein.
-     * @returns {boolean}
      */
-    getHasFadeIn() {
+    getHasFadeIn(): boolean {
         return true;
     }
 
     /**
      * Should return whether to fade out the game state. This will then apply the right css classes
      * for the fadeout and wait the delay before moving states
-     * @returns {boolean}
      */
-    getHasFadeOut() {
+    getHasFadeOut(): boolean {
         return true;
     }
 
     /**
      * Returns if this state should get paused if it does not have focus
-     * @returns {boolean} true to pause the updating of the game
+     * @returns true to pause the updating of the game
      */
-    getPauseOnFocusLost() {
+    getPauseOnFocusLost(): boolean {
         return true;
     }
 
     /**
      * Should return the html code of the state.
-     * @returns {string}
-     * @abstract
+     * @deprecated use {@link getContentLayout} instead
      */
-    getInnerHTML() {
-        abstract;
+    getInnerHTML(): string {
         return "";
     }
```

Comment:
maybe could use TypeScript abstract classes and methods?

Reply:
This is not abstract yet because there are still HTML string-based states and they're JS as well. Making even some methods abstract would force consumers to implement them, thus making legacy HTML strings support kinda pointless (the primary reason is to avoid rewriting states that we don't care about just yet, or those that would be rewritten later anyway).